sqlglot/parser.py

```diff
@@ -1834,6 +1835,9 @@ def extend_props(temp_props: t.Optional[exp.Properties]) -> None:
                 expression = self._parse_ddl_select()
 
             if create_token.token_type == TokenType.TABLE:
+                # source dialect
+                dialect = self.dialect.__module__.split(".")[-1].upper()
```

Comment:
This isn't the right way to do this, The AST is intentionally dialect-agnostic. Let's please remove this and solve the problem more idiomatically.

Reply:
Oh, I deleted it.